File: app/src/main/java/com/chris/eban/ViewModelModule.java
package com.chris.eban;

import com.chris.eban.common.EBanViewModelProviderFactory;
import com.chris.eban.domain.usecase.EventListQuery;
import com.chris.eban.domain.usecase.EventSaveInsert;
import com.chris.eban.domain.usecase.EventSaveUpdate;

import javax.inject.Singleton;

import androidx.lifecycle.ViewModelProvider;
import dagger.Module;
import dagger.Provides;

@Module
class ViewModelModule {

    @Singleton
    @Provides
    static ViewModelProvider.Factory providerFactory(EventListQuery eventListQuery,
                                                     EventSaveInsert eventSaveInsert,
                                                     EventSaveUpdate eventSaveUpdate) {
        return new EBanViewModelProviderFactory(eventListQuery, eventSaveInsert, eventSaveUpdate);
    }
}

File: app/src/main/java/com/chris/eban/ViewModelModule.kt
package com.chris.eban

import androidx.lifecycle.ViewModelProvider
import com.chris.eban.common.EBanViewModelProviderFactory

import com.chris.eban.domain.usecase.*

import com.chris.eban.domain.usecase.EventListQuery
import com.chris.eban.domain.usecase.EventSaveDelete
import com.chris.eban.domain.usecase.EventSaveInsert
import com.chris.eban.domain.usecase.EventSaveUpdate

import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ViewModelModule {

    @Singleton
    @Provides
    fun providerFactory(
            eventListQuery: EventListQuery,
            eventItemQuery: EventItemQuery,
            eventSaveInsert: EventSaveInsert,
            eventSaveUpdate: EventSaveUpdate,
            eventSaveDelete: EventSaveDelete
    ): ViewModelProvider.Factory {
        return EBanViewModelProviderFactory(
                eventListQuery,
                eventItemQuery,
                eventSaveInsert,
                eventSaveUpdate,
                eventSaveDelete
        )
    }
}
